### ISSUE TITLE
Getting MemberOf Groups for a particular user

### ISSUE THREAD

**Issue by tasgray** [microsoftgraph/msgraph-sdk-dotnet#611]
Hi there,

This is more so a question rather than an issue and perhaps I am missing something really obvious but in case I'm not, this will hopefully help others.

In the documention its very clear that to query the API using the GraphServiceClient to retrieve information about me using Me.Drive.Request().GetAsync(). The mapping between the API endpoint and the chaining makes perfect sense. However, where it falls over in my mind is in trying to get a list of groups a particular user is a member of. For example, /users/{id | userprincipalname}/memberOf (https://docs.microsoft.com/en-us/graph/api/user-list-memberof?view=graph-rest-1.0&tabs=http). I cannot find any methods in the GraphServiceClient which allow me to query that endpoint. 

In summary, my question is can the SDK do this in the manner I have tried or send a http request as shown in the documentation?

Many thanks

**Comment by zsybupt** [comment]
@tasgray  Me.MemberOf.Request().GetAsync()

**Comment by tasgray** [comment]
> @tasgray Me.MemberOf.Request().GetAsync()

Won't this give me the groups for the user I am authenticated with? If not, how do I specify which user to return the groups for?

**Comment by pschaeflein** [comment]
In general, questions should be posted to [Stack Overflow](https://stackoverflow.com/) and tagged with `microsoft-graph`.

From the [docs](https://docs.microsoft.com/en-us/graph/api/resources/users?view=graph-rest-1.0):

> You can access users through Microsoft Graph in two ways:
>    By their ID, /users/{id | userPrincipalName}
>    By using the /me alias for the signed-in user, which is the same as /users/{signed-in user's id}

If you do not know the id/upn, you could filter on displayName. (The docs have several examples of filtering users: https://docs.microsoft.com/en-us/graph/query-parameters?context=graph%2Fapi%2F1.0&view=graph-rest-1.0#filter-parameter)

**Comment by tasgray** [comment]
> In general, questions should be posted to [Stack Overflow](https://stackoverflow.com/) and tagged with `microsoft-graph`.
> 
> From the [docs](https://docs.microsoft.com/en-us/graph/api/resources/users?view=graph-rest-1.0):
> 
> > You can access users through Microsoft Graph in two ways:
> > By their ID, /users/{id | userPrincipalName}
> > By using the /me alias for the signed-in user, which is the same as /users/{signed-in user's id}
> 
> If you do not know the id/upn, you could filter on displayName. (The docs have several examples of filtering users: https://docs.microsoft.com/en-us/graph/query-parameters?context=graph%2Fapi%2F1.0&view=graph-rest-1.0#filter-parameter)

My apologies for that, I will go to Stack Overflow in future.

I managed to work out what I was looking for. In the documentation you only ever provide examples using 'me'. What I needed to do was use graphclient.Users[userid].MemberOf.Request().GetAsync(). There are no examples of that syntax in your documentation so may I suggest for the sake of others that you include one or two.

Many thanks
